src/envoy/http/authn/authn_utils.cc

```diff
@@ -22,51 +22,34 @@ namespace Http {
 namespace Istio {
 namespace AuthN {
 namespace {
-class JsonIterCallback : public Logger::Loggable<Logger::Id::filter> {
- public:
-  // The json iterator callback function
-  static bool JsonIteratorCallback(const std::string&,
-                                   const Envoy::Json::Object&,
-                                   istio::authn::JwtPayload* payload);
-};
+// The JWT audience key name
+static const std::string kJwtAudienceKey = "aud";
 
-bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
-                                            const Envoy::Json::Object& obj,
-                                            istio::authn::JwtPayload* payload) {
-  ENVOY_LOG(debug, "{}: key is {}", __FUNCTION__, key);
-  ::google::protobuf::Map< ::std::string, ::std::string>* claims =
-      payload->mutable_claims();
+// Extract JWT audience into the JwtPayload.
+// This function should to be called after
+// the claims are extracted into the payload.
+void ExtractJwtAudience(const Envoy::Json::Object& obj,
+                        istio::authn::JwtPayload* payload) {
+  const std::string& key = kJwtAudienceKey;
+  // "aud" can be either string array or string.
+  // First, try as string
+  if (payload->claims().count(key) > 0) {
```

Comment:
this code has dependency on the code in line 92.  it is better not to since it has "obj"
Please change this to use "obj" directly to check if "aud" as string.

Reply:
Done.